### PR TITLE
ci: build Tasklist and Operate frontends in distribution module

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description
This is to fix the issues found when running `release:perform` with concurrency, yarn install could not be run at the same time for Operate and Tasklist, thus we had to disable the concurrency for the releasing (which make it taking ~6 minutes longer)

In this pull request, `frontend-maven-plugin` executions are run by `dist` module to ensure that they are not run in parallel, it also allows that `install-node-and-yarn` goal is excuted only once.
A new profile `buildDistFrontend` was created to be used for mvn `release:perform` along with `skipFrontendBuild` that skips the same plugin executions in client submodules in Tasklist and Operate
The plugin is kept though in client modules, to allow developers to start the applications using webapp only (without building and running the single JAR) 

Dry-run for this branch https://github.com/camunda/zeebe/actions/runs/8971123064

## Related issues

closes #

### CODE REVIEW

File: .github/workflows/camunda-platform-release-main-dry-run.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
           - version: 0.8.2-alpha1
             latest: false
     with:
-      releaseBranch: main
```
Comment (houssain-barouni) [inline .github/workflows/camunda-platform-release-main-dry-run.yml:23]: needed to test dry-run, I will revert it

Comment (Zelldon) [reply .github/workflows/camunda-platform-release-main-dry-run.yml:23]: You could also use the other workflow https://github.com/camunda/zeebe/actions/workflows/camunda-platform-release-manual.yml and mark it as dryrun :)

Comment (houssain-barouni) [reply .github/workflows/camunda-platform-release-main-dry-run.yml:23]: indeed

File: .github/workflows/camunda-platform-release.yml [hunk 2]
```diff
@@ -181,7 +181,7 @@ jobs:
             -DlocalCheckout=${{ inputs.dryRun }} \
             -DskipQaBuild=true \
             -P-autoFormat \
-            -Darguments='-P-autoFormat -DskipQaBuild=true -DskipChecks=true -DskipTests=true -Dskip.fe.build=false -Dspotless.apply.skip=false -Dskip.central.release=${SKIP_REPO_DEPLOY} -Dskip.camunda.release=${SKIP_REPO_DEPLOY} -Dzbctl.force -Dzbctl.rootDir=${ZBCTL_ROOT_DIR} -Dgpg.passphrase="${{ steps.secrets.outputs.MAVEN_CENTRAL_GPG_SIGNING_KEY_PASSPHRASE }}"'
+            -Darguments='-T0.5C -P buildDistFrontend,skipFrontendBuild,-autoFormat -DskipQaBuild=true -DskipChecks=true -DskipTests=true -Dspotless.apply.skip=false -Dskip.central.release=${SKIP_REPO_DEPLOY} -Dskip.camunda.release=${SKIP_REPO_DEPLOY} -Dzbctl.force -Dzbctl.rootDir=${ZBCTL_ROOT_DIR} -Dgpg.passphrase="${{ steps.secrets.outputs.MAVEN_CENTRAL_GPG_SIGNING_KEY_PASSPHRASE }}"'
```
Comment (Zelldon) [inline .github/workflows/camunda-platform-release.yml:184]: ❌ That is a bit confusing :/ 

Could we either rename the profiles `skipFrontendBuild` or add a comment here why we doing this. See above I tried to document a bit the used parameter, might be useful also to explain why we have to use the two settings.

**Comment by houssain-barouni** [comment]
handled in https://github.com/camunda/zeebe/pull/18483

**Pull request closed without merging**